t/unit-tests/u-mem-pool.c

```diff
@@ -0,0 +1,28 @@
+#include "unit-test.h"
+#include "mem-pool.h"
+
+static void t_calloc_100(struct mem_pool *pool)
```

Comment:
None of the testcases do anything with their local pool. So how about we make this take as parameter the `block_alloc` size and have the function here handle the setup and teardown of the pool?